specs/_features/eip7594/p2p-interface.md

```diff
@@ -0,0 +1,211 @@
+
+*Note:* In the `verify_data_column_sidecar_inclusion_proof(sidecar)` check, for all the sidecars of the same block, it verifies against the same set of `kzg_commitments` of the given beacon block. Client can choose to cache the result of the arguments tuple `(sidecar.kzg_commitments, sidecar.kzg_commitments_inclusion_proof, sidecar.signed_block_header)`.
+
+### The Req/Resp domain
+
+#### Messages
+
+##### DataColumnSidecarsByRoot v1
+
+**Protocol ID:** `/eth2/beacon_chain/req/data_column_sidecars_by_root/1/`
+
+*[New in EIP7594]*
+
+The `<context-bytes>` field is calculated as `context = compute_fork_digest(fork_version, genesis_validators_root)`:
+
+[1]: # (eth2spec: skip)
+
+| `fork_version` | Chunk SSZ type |
+| - | - |
+| `EIP7594_FORK_VERSION` | `eip7594.DataColumnSidecar` |
+
+Request Content:
+
+```
+(
+  List[DataColumnIdentifier, MAX_REQUEST_DATA_COLUMN_SIDECARS]
+)
+```
+
+Response Content:
+
+```
+(
+    List[DataColumnSidecar, MAX_REQUEST_DATA_COLUMN_SIDECARS]
+)
+```
+
+Requests sidecars by block root and index.
+The response is a list of `DataColumnIdentifier` whose length is less than or equal to the number of requests.
+It may be less in the case that the responding peer is missing blocks or sidecars.
+
+Before consuming the next response chunk, the response reader SHOULD verify the data column sidecar is well-formatted, has valid inclusion proof, and is correct w.r.t. the expected KZG commitments through `verify_data_column_sidecar_kzg_proofs`.
+
+No more than `MAX_REQUEST_DATA_COLUMN_SIDECARS` may be requested at a time.
+
+The response MUST consist of zero or more `response_chunk`.
+Each _successful_ `response_chunk` MUST contain a single `DataColumnSidecar` payload.
+
+Clients MUST support requesting sidecars since `minimum_request_epoch`, where `minimum_request_epoch = max(finalized_epoch, current_epoch - MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS, EIP7594_FORK_EPOCH)`. If any root in the request content references a block earlier than `minimum_request_epoch`, peers MAY respond with error code `3: ResourceUnavailable` or not include the data column sidecar in the response.
+
+Clients MUST respond with at least one sidecar, if they have it.
+Clients MAY limit the number of blocks and sidecars in the response.
+
+Clients SHOULD include a sidecar in the response as soon as it passes the gossip validation rules.
+Clients SHOULD NOT respond with sidecars related to blocks that fail gossip validation rules.
+Clients SHOULD NOT respond with sidecars related to blocks that fail the beacon chain state transition
+
+### The discovery domain: discv5
+
+#### ENR structure
+
+##### `custody_subnet_count`
+
+A new field is added to the ENR under the key `custody_subnet_count` to facilitate custody data column discovery.
```

Comment:
Is this a MUST field or is the value assumed to be `MINIMUM` if it is not found there. and what if someone puts in the field and puts a value less than `MINIMUM`, e.g. 0.

Reply:
I would say it should be a MUST field. If the field is absent, the node should assume that the ENR is invalid according to the current fork.

Reply:
Not sure it makes a huge difference
However I'd prefer it to be optional and specify _extra_ subnet count. The way of finding voluntary nodes would be different from finding regular (obligatory) nodes. So putting there anything when a node doesn't commit to serving extra subnets looks a bit pointless to me.